905 fps fix

C-Prime90 opened the pull request:
- This fixes a issue were S905 would get stuck at 30.3 fps after video playback or some GUI transitions when using a 60fps resolution.
- This also adds a unique DTSI for S905X because S905 DTSI will result in kernel crashes on S905X.

C-Prime90 committed:
Revert "PD#125321: dtsi: gxbb: up the gpu limit"

This reverts commit f86325c2982a61bfbc4993d5eb200b45051599bb.

C-Prime90 committed:
GXL: Use unique Mali DTSI

This reverts commit 9c2c823513 for GXL devices

Ray-future commented:
@croniccorey: First of all thank you for this PR. Can you provide an example how this issue happens? I'd like to reproduce it and run some tests with this PR applied. Is the issue reproducible in Kodi?
So you are reverting the device tree change for S905 and you are creating a seperate DTSI with those non reverted values because S905X does not suffer from this issue? 

@kszaq do you have a S905X device to check?

kszaq commented:
S905X device tree that is proposed in this PR is something I've been using in my builds for a long time and looks good to me.

@croniccorey I would also like to know how can we reproduce the S905 issue that this PR is aiming to fix. Thank you.

C-Prime90 commented:
To reproduce the issue i seen on 905,  set resolution to 1080p60hz and enable debug logging (so fps is visible) after a number of gui transitions or video playback GUI FPS becomes stuck at 30.3 to fix you can reboot or change resolution but issue will return after time,  i tracked issue down to the commit i reverted for 905, i also added a separate dtsi for 905x because the 905 one with commit reverted will cause sporadic lock ups on 905x (ive tested this on 3 different 905x models).

905x still has fps issue that this commit fixes for 905,  only way ive managed to fix 905x FPS is to drastically lower clock witch is a poor solution IMO.

I heavily tested this commit with my buildroot before PR,  Once i have a proper fix for 905x i will PR it aswell.

Ray-future commented:
@croniccorey I'm not sure Kodi's debug OSD is good to measure that. I've asked Kodi Amlogic Developer @peak3d and he told me these values are not reliable. I quote: "According to Kodi, I'm playing my 50fps videos with 7fps". Memory consumption in Kodi OSD is also nonsense (98% on Odroid C2 all the time).
Is there any reliable way to check fps?

C-Prime90 commented:
regardless of osd reliability there was draging and sluggishness in GUI transitions witch led to me doing this,  after commit osd reported a stable 60.5 fps on homepage that previously was getting stuck at 30.3 fps and transitions became fluent and smooth.

the debug osd might not be reliable for video playback but it seems pretty accurate as far as gui goes.

Ray-future commented:
What mali are you using? We reverted to r5p1 in LibreELEC. I never saw the issue in LE you describe.

C-Prime90 commented:
We use R5P1, This issue was reported to us by our users and was reproducible on all my S905 platforms, it was very noticeable when transitioning (hitting back button from a sub menu back to home screen would cause drag almost like a slow fading effect).

C-Prime90 commented:
more testing has narrowed the issue down to frequencies below 400mhz (125mhz,  250mhz,  285mhz),  this is the reason you are not seeing it in the LE kernel as it sets minimum frequency above this.

if you set the max frequency in '/sys/class/mpgpu'  lower then 400mhz you should see the issue show on kodi.

chewitt commented:
LE folks can I ask where this is at? Good to merge?

Ray-future commented:
@chewitt I'm still not able to reproduce the issue. @kszaq reverted 9c0ad98 and even with 285mhz there is no issue.
@croniccorey: Can you test https://github.com/LibreELEC/LibreELEC.tv/pull/1390?

kszaq commented:
I was also not able to reproduce this with S905 nor S905X.

chewitt commented:
I'm closing this due to lack of activity. Thanks.
